documentation/CAMARA-AuthN-AuthZ-Concept.md

```diff
@@ -136,6 +136,15 @@ It enables Clients to verify the identity of the End-User based on the authentic
 <img src="./images/oidc.png" alt="OIDC"
 	title="Abstarct OIDC Flow" width="350" height="300" />	
 
+ OIDC provides a framework for identity providers to support a wide range of authentication methods/factors that can be used to verify the identity of the user:
+
+ - Username and Password: The user provides the client application with a username and password. The client application then sends the username and password to the identity provider, which verifies the user's identity and returns an ID token to the client application.
+ - SMS One-Time Password (OTP): The user provides the client application with a phone number. The client application then sends the phone number to the identity provider, which sends an SMS message to the user's phone number. The user enters the code from the SMS message into the client application, which sends the code to the identity provider. The identity provider verifies the code and returns an ID token to the client application.
+- Network-based authentication: An authentication mechanism based on the identification of the endpoint of a network connection (e.g., by the observed IP address of the user device connected to the network). Network operators know which user is associated with a network resource at any given time, such as the mobile phone number associated with a particular cellular connection.<br>This is particularly interesting in the context of CAMARA APIs exposed by network operators. For example, it is the authentication method defined for the [Number Verification API](https://github.com/camaraproject/NumberVerification) without any user interaction in the process (aka "silent" auth).
```

Comment:
Hi! I have some questions to this one:
1. How do developers know what network auth solution works in particular MNO?
2. How does CAMARA want to bring this knowledge and technical flow to this complexity? (I can name at least 4 solutions inside DT... not mentioning how it works worldwide...)

Reply:
Hi @DT-DawidWroblewski. 
1. No need to know as developer just needs network auth to happen. They trigger the standard AuthCode flow, which is based on http redirects and network auth happens there. It could be auth based on observed public IP or on other mechanisms (notice definition includes use of observed IP as an example, not as the unique valid solution)

2. Why do you think CAMARA should standardize this? It does not matter if e.g. Telefonica uses mapping of public IP to MSISDN while DT uses another mechanism, as long as it fits the concept defined as Network-based authentication as defined above. In Telefonica it similar to DT, we don't have the same exact mechanism in every country

Reply:
Hi @diegogonmar

1. As long as customer is aware what is going on behind /authorize - it is fine. The only concern I have is that statement **Network-based authentication** might be confusing, e.g. when one MNO use HE and it requires developer to send a request over HTTP (HE does not with HTTPS; how it would work for Web and solutions like *Apple Private Relay*?), another MNO use *redirect* mechanism, so it requires developer to implement a proper software code to handle redirects and on top of this, redirects might use different flows behind, e.g. simple request response or oidc/oauth solutions that involves *code grant* flow. Developer needs to know how to handle particular solution, otherwise she or he gets stuck on proper handling of **Network-based authentication**. 
2. Camara doesn't need to specify this and I think that it should deliver the most valuable information about Telco APIs back to developer.

Reply:
Hi @DT-DawidWroblewski,
Regarding 1:
- As I indicated before it will be transparent for developer. OAuth2 AuthCode flow is based on HTTP requests and involves redirections. Need to handle redirects is not a requirement or something to be known, its something basic attached to the fact of using Authorization Code. OAuth RFC says: _This specification makes extensive use of HTTP redirections, in which the client or the authorization server directs the resource owner's user-agent to another destination_ (https://www.rfc-editor.org/rfc/rfc6749#section-1.7) and regarding authorization code grant it says _this is a redirection-based flow_ (https://www.rfc-editor.org/rfc/rfc6749#section-4.1)
So, if one MNO makes a redirect to an endpoint where HE is working its an option, if another sees public IP is a different option. In both options any further redirects may occur (as it usually happen)
- Also notice that Network Auth is only applied to Authorization Code flow, so I don't think what you say about request/response applies, it will always be based on redirects as it said by standard
- Finally, notice in no place its said that this is a universal and always valid solution. This is one of the solutions availabie that may apply or not to the APIs depending on conditions. To go to the obvious case, it won't work if there is no Mobile Network connection, Network-based auth will not be possible. 
Regarding 2 we are alligned, CAMARA won't specify specific mechanisms for Network Based Auth, that is not relevant for developer